File: array/appear_once.js
var arr = [1, 1, 2, 2, 3, 4, 5, 5];

var set = new Set();
var once = new Set();

for (let i of arr) {
    if (!set.has(i)) {
        if (!once.has(i)) {
            set.add(i);
            once.add(i);
        }
    } else {
        if (once.has(i)) {
            once.delete(i);
        }
    }
}

console.log(once);

File: array/appear_once.py
arr = [1, 1, 2, 2, 3, 4, 5, 5]

s, o = set(), set()

for i in arr:
    if not i in s:
        if not i in o:
            s.add(i)
            o.add(i)
    else:
        if i in o:
            o.remove(i)

print(o)
